elasticsearch/templates/statefulset.yaml

```diff
@@ -215,10 +215,17 @@ spec:
             {{- end }}
           {{- end }}
           {{- range $path, $config := .Values.esConfig }}
-          - name: esconfig
+          - name: config
             mountPath: /usr/share/elasticsearch/config/{{ $path }}
             subPath: {{ $path }}
           {{- end -}}
+  {{- if .Values.hooks }}
+          {{- range $path, $config := .Values.hooks }}
+          - name: config
+            mountPath: /{{ $path }}
```

Comment:
This is just identical to the existing block above it, with the only difference that is is being mounted into the root directory instead of the elasticsearch config directory. Would you have any issue with just mounting the files via `esConfig`? Just removes the code duplication that does exactly the same thing but in a different directory.

Reply:
@Crazybus here is used different variables in `range`, do you suggest smth like:
```
{{- range $path, $config := .Values.esConfig }}
- name: config
   mountPath: /usr/share/elasticsearch/config/{{ $path }}
  subPath: {{ $path }}
{{- end -}}
{{- range $path, $config := .Values.hooks }}
- name: config
   mountPath: /usr/share/elasticsearch/config/{{ $path }}
    subPath: {{ $path }}
{{- end -}}
```
but anyhow here will be duplication. or?

Reply:
The same thing could be said about putting this under `hooks`. This is actually adding functionality for mounting a file into the container (the same thing that `esConfig` does) and has nothing to do with hooks. If we need a generic way to mount files into the container that are outside `esConfig` then a new more generic variable should be added to do that like `files. 

A hook specific one is also not even needed for this use case, instead you can just put the script inline directly in the lifecycle section which is what is documented in the Kubernetes examples:

```
lifecycle:
  postStart:
    exec:
      command: ["/bin/sh", "-c", "echo Hello from the postStart handler > /usr/share/message"]
```

And if you wanted to do this with a multiline script it would look like:

```
lifecycle:
  postStart:
    exec:
      command:
        - sh
        - -c
        - |
          #!/bin/bash
          set -euo pipefail
          echo Hello from the postStart handler > /usr/share/message
          echo this is a multiline script!
```

Reply:
@Crazybus thank you for the clarification! it totally makes sense. I've removed `hooks`, kept only `lifecycle` related changes. Could you please have a second look?